src/lib/github_old.ts

```diff
@@ -0,0 +1,176 @@
+// };
+
+// export function getEventData(): PREventData {
+//   const eventPath = process.env.GITHUB_EVENT_PATH ?? '';
+//   const eventData = JSON.parse(readFileSync(eventPath ?? '', 'utf8'));
+//   // console.log('\neventData\n', stringify(eventData));
+//   if (!eventData) {
+//     core.setFailed(`Event data not found: ${eventPath}`);
+//     process.exit(1);
+//   }
+//   return eventData as PREventData;
+// }
+
+// export type PRDetails = {
+//   action: 'opened' | 'closed' | 'reopened' | 'synchronize';
+//   owner: string;
+//   repo: string;
+//   pullNumber: number;
+//   title: string;
+//   description: string | null;
+//   baseSha?: string; // before
+//   headSha?: string; // after
+// };
+
+// export async function getPRDetails(): Promise<PRDetails> {
+//   const token = getGithubToken();
+//   const { repository, action, number } = getEventData();
+
+//   core.notice(`PR Event: ${action}`);
+
+//   const params = {
+//     owner: repository.owner.login,
+//     repo: repository.name,
+//     pull_number: number
+//   };
+
+//   const octokit = github.getOctokit(token);
+//   const response = await octokit.rest.pulls.get(params);
+
+//   if (!response) {
+//     core.setFailed('PR details not found');
+//     process.exit(1);
+//   }
+
+//   const result: PRDetails = {
+//     action: event.action,
+//     description: response.data.body,
+//     owner: event.repository.owner.login,
+//     pullNumber: event.number,
+//     repo: event.repository.name,
+//     title: response.data.title,
+//     baseSha: event?.before || response?.data?.base?.sha,
+//     headSha: event?.after || response?.data?.head?.sha
+//   };
+
+//   return result;
+// }
+
+// export async function getDiff(owner: string, repo: string, pullNumber: number): Promise<string | null> {
+//   const token = getGithubToken();
+//   const octokit = github.getOctokit(token);
+
+//   const response = await octokit.rest.pulls.get({
+//     owner,
+//     repo,
+//     pull_number: pullNumber,
+//     mediaType: { format: 'diff' }
+//   });
+//   // @ts-expect-error - response.data is a string
+//   return (response?.data || '') as string;
+// }
+
+// export async function compareCommits({ owner, repo, baseSha, headSha }: PRDetails) {
+//   const token = getGithubToken();
+
+//   if (!baseSha || !headSha) return null;
+
+//   const octokit = github.getOctokit(token);
+//   const response = await octokit.rest.repos.compareCommits({
+//     headers: { accept: 'application/vnd.github.v3.diff' },
+//     owner,
+//     repo,
+//     base: baseSha,
+//     head: headSha
+//   });
+
+//   // @ts-ignore
+//   return (response?.data || '') as string;
+// }
+
+export type Comment = {
+  body: string;
+  path: string;
+  line: number;
+};
+export async function createReviewComment({ owner, repo, pullNumber }: PRDetails, comments: Comment[]) {
+  const token = getGithubToken();
+  const octokit = github.getOctokit(token);
+  const filename = comments?.[0]?.path;
+
```

Comment:
⚠️
A variável `filename` pode resultar em `undefined` se `comments` for um array vazio. Acessar `[0]` diretamente pode levar a um erro.
É importante verificar se o array `comments` não está vazio antes de acessar seus elementos para prevenir erros de tempo de execução.